### PR TITLE
Added action check for game_server

### PR DESCRIPTION
**Pull request by bronvic**
Added `check_action_available` function which checks if user can perform there action

Solves #124

### CODE REVIEW

File: backend/src/game_server.rs [hunk 1]
```diff
@@ -57,16 +108,21 @@ impl Handler<inner::Request<Click>> for GameServer {
...
+                        Action::Select => {
+                            // This arm is for selection and reselsection
+                            // If we have no unit selected, we can safely call deselect
+                            self.deselect_unit();
+                            self.select_unit(click.target)
+                        }
+                        Action::Move => self.move_unit(click.target),
+                        Action::Attack => self.attack_unit(click.target),
+                    },
+                    false => Err(eyre!("action anavalable")),
```
Comment (Hairash) [inline backend/src/game_server.rs:124]: "unavailable"

File: backend/src/game_server.rs [hunk 2]
```diff
@@ -36,6 +36,57 @@ impl GameServer {
...
+           Action::Select | Action::Deselect => {
+               // if player does not have any selections, they can select and deselect
+               let hex = match self.game.selected_hex {
+                   Some(hex) => hex,
+                   None => return true,
+               };
+
+               let unit = match hex.get_unit() {
+                   Some(unit) => unit,
+                   None => return false,
```
Comment (Hairash) [inline backend/src/game_server.rs:56]: should be error

File: backend/src/game_server.rs [hunk 3]
```diff
@@ -36,6 +36,57 @@ impl GameServer {
...
+           },
+           Action::Move => {
+               let hex = match self.game.selected_hex {
+                   Some(hex) => hex,
+                   None => return false,
+               };
+
+               let unit = match hex.get_unit() {
+                   Some(unit) => unit,
+                   None => return false,
```
Comment (Hairash) [inline backend/src/game_server.rs:70]: error again

File: backend/src/game_server.rs (hunk 2 above)
Comment (Hairash) [reply backend/src/game_server.rs:56]: Repeated block with assigning hex and unit variables.
Better to move above.

**Review by Hairash — CHANGES_REQUESTED** [review]

File: backend/src/game_server.rs (hunk 1 above)
Comment (bronvic) [reply backend/src/game_server.rs]: yep

File: backend/src/game_server.rs (hunk 2 above)
Comment (bronvic) [reply backend/src/game_server.rs]: you will have to check if they none or some even if you get them above

Comment (bronvic) [reply backend/src/game_server.rs]: not sure should the error be here: need to discuss

**Review by Hairash — APPROVED** [review]
good code!

**Comment by Hairash** [comment]
close #124

**Comment by Hairash** [comment]
close #129

**Pull request merged**